### PR TITLE
Master website twitter new api rop

### PR DESCRIPTION
**Pull request by rop-odoo**
Task: https://www.odoo.com/web?#id=12130&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.frrrQC7LwpPCgnNG

### CODE REVIEW

File: addons/website_twitter/controllers/main.py [hunk 1]
```diff
@@ -11,31 +14,29 @@ def twitter_reload(self):
...
-        secret = request.website.twitter_api_secret
-        screen_name = request.website.twitter_screen_name
-        cr, uid = request.cr, request.uid
-        debug = request.registry['res.users'].has_group(cr, uid, 'base.group_website_publisher')
-        if not key or not secret:
-            if debug:
+        Key = request.website.twitter_api_key
+        Secret = request.website.twitter_api_secret
+        ScreenName = request.website.twitter_screen_name
+        Debug = request.env['res.users'].has_group('base.group_website_publisher')
```
Comment (rga-odoo) [inline addons/website_twitter/controllers/main.py]: recheck all variable name according to [guideline](https://www.odoo.com/documentation/master/reference/guidelines.html), use camelcase only for model variable

File: addons/website_twitter/controllers/main.py [hunk 2]
```diff
@@ -11,31 +14,29 @@ def twitter_reload(self):
...
+        if len(Tweets) < 12:
+            if Debug:
                 return {"error": _("Twitter user @%(username)s has less than 12 favorite tweets. "
                                    "Please add more or choose a different screen name.") % \
-                                      {'username': screen_name}}
+                                      {'username': ScreenName}}
             else:
                 return []
-        return [json.loads(tweet['tweet']) for tweet in tweets]
+        return [json.loads(Tweet['tweet']) for Tweet in Tweets]
```
Comment (rga-odoo) [inline addons/website_twitter/controllers/main.py]: Tweets.mapped('tweet')

File: addons/website_twitter/models/website_twitter.py [hunk 3]
```diff
@@ -41,50 +40,44 @@ def _request(self, website, url, params=None):
...
-            last_tweet = website_tweets.search_read(
-                    cr, uid, [('website_id', '=', website.id),
-                              ('screen_name', '=', website.twitter_screen_name)],
-                    ['tweet_id'],
-                    limit=1, order='tweet_id desc', context=context)
+            last_tweet = WebsiteTweets.search([('website_id', '=', website.id),
+                                                     ('screen_name', '=', website.twitter_screen_name)],
+                                                     limit=1, order='tweet_id desc')
             if last_tweet:
                 params['since_id'] = int(last_tweet[0]['tweet_id'])
```
Comment (rga-odoo) [inline addons/website_twitter/models/website_twitter.py]: last_tweet[0]['tweet_id'] ==> last_tweet.tweet_id

File: addons/website_twitter/controllers/main.py [hunk 4]
```diff
@@ -14,28 +17,26 @@ def get_tweets(self, limit=20):
         key = request.website.twitter_api_key
         secret = request.website.twitter_api_secret
         screen_name = request.website.twitter_screen_name
-        cr, uid = request.cr, request.uid
-        debug = request.registry['res.users'].has_group(cr, uid, 'base.group_website_publisher')
+        Debug = request.env['res.users'].has_group('base.group_website_publisher')
```
Comment (rga-odoo) [inline addons/website_twitter/controllers/main.py]: Debug ==> debug

File: addons/website_twitter/controllers/main.py [hunk 5]
```diff
@@ -14,28 +17,26 @@ def get_tweets(self, limit=20):
...
         if len(tweets) < 12:
-            if debug:
+            if Debug:
                 return {"error": _("Twitter user @%(username)s has less than 12 favorite tweets. "
                                    "Please add more or choose a different screen name.") % \
                                       {'username': screen_name}}
             else:
                 return []
-        return [json.loads(tweet['tweet']) for tweet in tweets]
+        return [json.loads(tweet) for tweet in tweets.mapped('tweet')]
```
Comment (rga-odoo) [inline addons/website_twitter/controllers/main.py]:
```python
tweets.mapped(lambda t: json.loads(t.tweet))

File: addons/website_twitter/models/website_twitter.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            return json.load(urlopen(request, timeout=URLOPEN_TIMEOUT))
+        except HTTPError, e:
+            _logger.debug("Twitter API request failed with code: %r, msg: %r, content: %r",
+                          e.code, e.msg, e.fp.read())
+            raise
+
+    @api.model
+    def _refresh_favorite_tweets(self):
+        ''' called by cron job '''
+        Website = self.env['website'].search([('twitter_api_key', '!=', False),
```
Comment (rga-odoo) [inline addons/website_twitter/models/website_twitter.py]: Website ==> website

File: addons/website_twitter/__openerp__.py [hunk 7]
```diff
@@ -12,9 +15,9 @@
...
         'security/ir.model.access.csv',
-        'data/twitter_data.xml',
-        'views/twitter_view.xml',
-        'views/twitter_snippet.xml'
+        'data/website_twitter_data.xml',
+        'views/website_twitter_settings_views.xml',
+        'views/website_twitter_snippet_templates.xml'
     ],
     'demo': [],
     'qweb': [],
```
Comment (WHPODOO) [inline addons/website_twitter/__openerp__.py]: Not use then why put here ? Remove it
```
'demo': [],
'qweb': [],
```

File: addons/website_twitter/models/website_twitter.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+                          e.code, e.msg, e.fp.read())
+            raise
+
+    @api.model
+    def _refresh_favorite_tweets(self):
+        ''' called by cron job '''
+        website = self.env['website'].search([('twitter_api_key', '!=', False),
+                                          ('twitter_api_secret', '!=', False),
+                                          ('twitter_screen_name', '!=', False)])
+        _logger.debug("Refreshing tweets for website IDs: %r", website.ids)
```
Comment (WHPODOO) [inline addons/website_twitter/models/website_twitter.py]: here need translation

Comment (msh-odoo) [inline addons/website_twitter/models/website_twitter.py]: @WHPODOO  who translates logger ?

Comment (WHPODOO) [inline addons/website_twitter/models/website_twitter.py]: @msh-odoo first look the code and then comment , check line no ```56``` in very next file ```website_twitter_config.py```

**Pull request merged**